scripts: use `grep -E` instead of `egrep`

a1346054 opened the pull request:
latest GNU grep emits a deprecation message when `egrep` is invoked. All platforms that can be expected to run the three changed scripts already have `grep -E` as a replacement.

emanuele6 commented:
There is no reason to use `-E` with these `grep` invocations since you are using `grep` and not `egrep`.

emanuele6 commented on MacOSX-Framework:
```diff
@@ -87,8 +87,8 @@ then
  #We "know" that 10.4 and earlier do not support 64bit
- OLD_SDK64=`ls  $SDK_PATH|egrep -v "10.[0-4]"|head -1`
- NEW_SDK64=`ls -r $SDK_PATH|egrep -v "10.[0-4][^0-9]" | head -1`
+ OLD_SDK64=`ls  $SDK_PATH|grep -Ev "10.[0-4]"|head -1`
+ NEW_SDK64=`ls -r $SDK_PATH|grep -Ev "10.[0-4][^0-9]" | head -1`
```
you don't need to use `-E` here

a1346054 replied:
I didn't want to change the script more than I had to.

vszakats replied:
Strictly speaking, `.` should be escaped here as well. This, and dropping `-E` are fine within this PR IMO.

We have more `egrep`s in this file:
```
.github/workflows/hacktoberfest-accepted.yml:            egrep -i "^Close[sd]? " | sort | uniq | tee log
.github/workflows/hacktoberfest-accepted.yml:          egrep -o "#([0-9]+)" log | cut -d# -f2 | sort | uniq | xargs -t -n1 -I{} \
.github/workflows/hacktoberfest-accepted.yml:            egrep -o '^([0-9]+):[0-9]{4}-(09-30T|10-|11-01T)' | cut -d: -f1 | sort | uniq | xargs -t -n1 -I {} \
.github/workflows/hacktoberfest-accepted.yml:          egrep -o "github.com/(.+)/(.+)/pull/([0-9]+)" log | sort | uniq | xargs -t -n1 -I{} \
.github/workflows/hacktoberfest-accepted.yml:            egrep -o '^([0-9]+):[0-9]{4}-(09-30T|10-|11-01T)' | cut -d: -f1 | sort | uniq | xargs -t -n1 -I {} \
```

a1346054 replied:
Addressed your comments in the fixed up commit.

a1346054 commented:
Fixed up the commit by removing `-E` where not needed for the grep pattern, and additionally escaping the `.` in the pattern.

Also replaced `egrep` in hacktoberfest-accepted.yml with `grep -E`

bagder commented:
Thanks!
